### ISSUE TITLE
Add helper config to deal with field collections.

### ISSUE THREAD

**Issue by e0ipso** [RESTful-Drupal/restful#613]
If you have something like:

`Article (node:article)` :arrow_right: `Guest (field_collection_item:field_article_guest)` :arrow_right: `Person (node:person), Collaboration type (taxonomy_term:colaboration), …`.

You want to be able to include the elements inside of the field collection directly in the article. So you could get the article's guest person node, the article's guest collaboration type, …

I'm proposing to have:

``` php
$public_fields['collaborator'] = array(
  'property' => 'field_guest_person',
  'brigde' => array(
    'property' => 'field_article_guest',
    // We can derive a lot of information about field collections automatically. If the
    // information is not bridged through a field collection, add here the 'entityType',
    // … pointed by 'field_article_guest'.
  ),
);
$public_fields['collaborationType'] = array(
  'property' => 'field_guest_collaboration_type',
  'brigde' => array(
    'property' => 'field_article_guest',
  ),
);
```

This way you don't need to write _stupid_ resources like `ArticleGuestFieldCollection__1_0` to embed your relationships/fields.

@amitaibu, @thebruce thoughts?

**Comment by e0ipso** [comment]
Another example could be:

`Video` :arrow_right: `field collection` :arrow_right: `Show relationship, Season relationship, Episode relationship`.

**Comment by thebruce** [comment]
If this is about can vs has to - then I think I am for it - I guess there a two lines of thought here and I can kind of see both: 

An Entity based theory on API Design
1) we might want to represent all entities regardless because it is good to have a pattern-able approach (http methods GET, PUT, POST/PATCH, DELETE) against all entities that exist in our API

OR

A State transition theory on API Design
2) APIs are about managing state, if we don't intend to expose any state management for an entity then we needn't/shouldn't represent it as an _embed and should not worry about HATEoas operations against it.

I think I probably lean more to the second line of thinking, but I can see the perspective of the first one. So I think if this is -- you have a choice, then I am all for it.

**Comment by amitaibu** [comment]
> This way you don't need to write stupid resources like ArticleGuestFieldCollection__1_0 to embed your relationships/fields.

I agree it would be silly to expose it as an endpoint, but maybe it makes sense to declare it still, just don't provide  `hook_menu()` for it. The reason is that if you'll do `$wrapper->field_guest_person-> field_article_guest->value()` you very well might get lots of Drupalism leaked in (e.g. text fields with their `format`, image fields with weird arrays, and entity reference that have no `full_view`).

So maybe a more robust, and _consistent_ way is to have a `$this->disableRouterItem = TRUE`, that keeps the resource, but doesn't declare hook_menu for it. (or we could just use the existing `$plugin['menu_item'] = FALSE`)
